File: Python/Graph.py
from tabulate import tabulate
class Graph:
    def __init__(self,edges):
        self.edges = edges
        self.graph_dict={}
        for (start,end) in self.edges:
            if start in self.graph_dict:
                self.graph_dict[start].append(end)
            else:
                self.graph_dict[start] = [end]
        print(self.graph_dict)

    def get_paths(self,start,end,paths=[]):
        paths = paths+[start]

        if start == end:
            return [paths]
        
        if start not in self.graph_dict:
            return []
        
        path=[]
        for node in self.graph_dict[start]:
            if node not in paths:
                new_paths = self.get_paths(node,end,paths)
                for p in new_paths:
                    path.append(p)
        return path

routes  =[
    ('Mumbai','Paris'),
    ('Mumbai' , 'Dubai'),
    ('Paris','Dubai'),
    ('Paris','New York'),
    ('Dubai','New York'),
    ('New York','Toronto'),
    ('Mumbai','Toronto'),
]

route_graph = Graph(routes)
print()
print(tabulate(route_graph.get_paths("Mumbai","New York")))

File: CPP/Graph.cpp
#include <bits/stdc++.h>

using namespace std;

/*class Graph{
private:
	vector<pair<string,string>> routes;
public:
	unordered_map<string,vector<string>> graph_map;
	Graph(vector<pair<string,string>> arr){
		this->routes = arr;
	}
	for(int i=0;i<this->routes.size();i++){
		
	}
};*/


int main(int argc, char const *argv[])
{
	#ifndef ONLINE_JUDGE
	 freopen("input.txt","r",stdin);	
	 freopen("output.txt","w",stdout);	
	#endif

	cout<<"Enter the number of routes : ";
	int n;cin>>n;
	cout<<n<<"\n";

	vector<pair<string,string>> routes;
	for (int i=0;i<n;i++){
		pair<string,string> route ;
		string start,end;
		cin>>start>>end;

		route.first = start;
		route.second = end;

		routes.push_back(route);
	}

	// vector<pair<string,string>> :: iter;

	for (int i=0;i<n;i++){
		cout<<i+1<<".  "<<routes[i].first<<" to "<<routes[i].second<<"\n";
	}

	return 0;
}

/*
5
Mumbai Paris
Mumbai Dubai
Paris  LA
Dubai  LA
Duabi  Paris
*/